src/expr/src/vector_op/cast.rs

```diff
@@ -274,43 +273,6 @@ where
         .map_err(|_| ExprError::Parse(type_name::<T>().into()))
 }
 
-// Define the cast function to primitive types.
-//
-// Due to the orphan rule, some data can't implement `TryFrom` trait for basic type.
-// We can only use [`ToPrimitive`] trait.
-//
-// Note: this might be lossy according to the docs from [`ToPrimitive`]:
-// > On the other hand, conversions with possible precision loss or truncation
-// are admitted, like an `f32` with a decimal part to an integer type, or
-// even a large `f64` saturating to `f32` infinity.
```

Comment:
The orphan rule is no longer a problem because we have custom `F32` / `F64` type. And we are handling the edge cases in the same way as PostgreSQL cast rather than `ToPrimitive` or rust `as` operator.